scipy/stats/_qmc.py

```diff
@@ -22,10 +36,30 @@
     _cy_wrapper_update_discrepancy
 )
 
+
 __all__ = ['scale', 'discrepancy', 'update_discrepancy',
            'QMCEngine', 'Sobol', 'Halton', 'LatinHypercube',
            'MultinomialQMC', 'MultivariateNormalQMC']
 
+IntNumber = Union[int, np.integer]
+DecimalNumber = Union[float, np.floating]
```

Comment:
```suggestion
DecimalNumber = Union[float, np.floating, np.integer]
```
Note that `builtins.int` is considered a subtype of `builtins.float` (by static type checkers, that is), 
but the same does not hold true for `np.integer` and `np.floating`.
I'd suggest making this change, unless you very explicitly want to exclude `np.integer`.